### ISSUE TITLE
Can't get "multiple practice mode" to show up in UI

### ISSUE THREAD

**Issue by gitouttatown** [OSEHRA/popHealth#94]
In popHealth.yml I have:

  #use practice-based system instead of provider opml hierarchy
  use_opml_structure: false

I reloaded the bundle file and restarted popHealth with 
sudo stop delayed_worker
sudo start delayed_worker

But when logged in as Administrator I don't see anything about "practices" anywhere in the PopHealth UI. 
What did I miss?
Thanks,

**Comment by gitouttatown** [comment]
About 30 minutes later, the "Practices" menu item showed up.
Is there some caching going on with regard to settings?
Thanks,

**Comment by petercyli** [comment]
There are asset caching which may need to be cleared for the UI to get updated after change in the configuration.

One thing I should mention, the system doesn't work right if you started working with patient data in non-multiple-practice mode and then switch to multi-practice mode after.  The fields within the patient "record" collection are define differently between regular and multiple-practice mode.  It is best to clear the patient data and associated patient cache and re-import all your data when you switch modes.

**Comment by gitouttatown** [comment]
Yes, I should have mentioned: I also deleted patients, deleted providers and deleted caches, before reloading the 2016 measures bundle and restarting.

When you say there is asset caching - is it something more than the deleting the caches via the "delete caches" button on the Patients page?

The odd thing is that the change did not appear right after restarting, but sometime later.
Similarly, I have turned on the "Baseline" option 
 `show_measure_baseline_report: true `
but there is no button showing up for the baseline feature 
Does some additional file have to exist?
I've changed the title of this topic because I seem to be encountering some sort of delayed action with configuration changes, not just the issue of enabling "Practices"

**Comment by petercyli** [comment]
I had same issue with inability to display the Baseline button.  

Luke suggested using the following commands to clear the cache:
rake tmp:clear 
rake assets:clobber

When I did that, the UI lost all of its css/javascript files.  I ran the "bower install" and "rake assets:precompile" to get the UI to display correctly.

**Comment by gitouttatown** [comment]
I did all that, but still no "benchmark" button displays.
Isn't it supposed to be somewhere near here:
![image](https://cloud.githubusercontent.com/assets/15641861/25668094/de50119c-2fe2-11e7-9982-b87a083a41d0.png)

Also, what are those two little icons to the left of the "Export" button for? They don't seem to do anything and have no tooltips or other clues about their purpose.

Thanks again.

**Comment by petercyli** [comment]
Yes. It should next to the "EXPORT" button.  It seems the "alignment" buttons doesn't work anymore.  Curious why it was created before.  

Anyway, I presume you ran the command - rake import:measure_baselines[test/fixtures/measure_baselines.json,true] RAILS_ENV=production.  Please make sure to use a proper measure id and correct start and end time that falls within the selected reporting period.  The example json file uses a bogus measure id and arbitrary start and end time.

You should be able to see the button.

**Comment by gitouttatown** [comment]
Thanks Peter. I didn't run that yet - that's what I was wondering about when I asked earlier if an additional file is needed. Now I see the json file for the baselines, and maybe I can figure that out.

I only just now found the documentation for baselines at 
[https://github.com/OSEHRA/popHealth/wiki/Static-Baseline-Report-Setup](url)
...so I have some reading to do before I pester anyone with more questions.
Newbies... sheesh.

**Comment by lrasmus** [comment]
@gitouttatown ask away, we're happy to help :)
@petercyli I updated https://github.com/OSEHRA/popHealth/wiki/Static-Baseline-Report-Setup to include the steps about clearing the rails cache assets

**Comment by dabimahesh** [comment]
Hi

I am using Pophealthv6.22 (latest release). I am trying to create multi-practice scenario.

1. I created one new practice
2. Registered one user and assigned to that practice with Staff Role
3. Uploaded patients for practice.
4. When I login as this new user I am unable to see any providers which are uploaded for that practice. That's why I am unable to see the dashboard
5. Also I am unable to upload patients with only staff role. I have to give admin role to this user. But with admin role it is not restricted to it's own practice, it shows data across all practices (same as default use pophealth.

So question is how to configure practice level segregation?

Thanks
Mahesh


Thanks
Mahesh
